src/main/java/com/fasterxml/jackson/databind/ser/PropertyBuilder.java

```diff
@@ -182,8 +182,12 @@ protected BeanPropertyWriter buildWriter(SerializerProvider prov,
             }
             if (valueToSuppress == null) {
                 suppressNulls = true;
-                // [databind#4464] NON_DEFAULT does not work with NON_EMPTY for custom serializer
-                valueToSuppress = BeanPropertyWriter.MARKER_FOR_EMPTY;
+                // [databind#4471] Different behavior when Include.NON_DEFAULT
+                //   setting is used on POJO vs global setting, as per documentation.
+                if (!_useRealPropertyDefaults) {
```

Comment:
Instead of re-checking configuration settings, let's rely on "_useRealPropertyDefaults" which means that POJO settings are in use (not global-, per-type or property)?

Reply:
Makes sense 👌🏼